pkgs/applications/virtualization/kata-containers/default.nix

```diff
@@ -0,0 +1,196 @@
+{ stdenv, buildGoPackage, fetchFromGitHub, fetchurl, pkgconfig, libseccomp, go
+, qemu }:
+
+let
+  version = "1.11.2";
+
+  # prepare source to have a structure of GOPATH
+  setSourceRoot = ''
+    mkdir -p "go/src/$(dirname "$goPackagePath")"
+    cp -R source "go/src/$goPackagePath"
+    export sourceRoot="go/src/$goPackagePath"
+  '';
+
+  # default make flags, set GOPATH, GOCACHE and install prefix
+  makeFlags = [
+    "GOPATH=$(NIX_BUILD_TOP)/go"
+    "GOCACHE=$(TMPDIR)/go-cache"
+    "PREFIX=$(out)"
+  ];
+
+  # default meta
+  meta = with stdenv.lib; {
+    homepage = "https://katacontainers.io/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ offline ];
+    platforms = platforms.linux;
+  };
+
+  agent = stdenv.mkDerivation {
+    pname = "kata-agent";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "kata-containers";
+      repo = "agent";
+      rev = version;
+      sha256 = "6Zr7qwBuidJmKdZL2VGmRlKthWletdc0fRS5kA54CHM=";
+    };
+
+    goPackagePath = "github.com/kata-containers/agent";
+
+    inherit setSourceRoot;
+
+    nativeBuildInputs = [ pkgconfig go ];
```

Comment:
is this a go package? I would probably use `buildGoPackage`, seeing as they still haven't converted to using go modules

Reply:
It won't be easy to make this work with with our go builders.

Reply:
v2 is in alpha, it switches to monorepo so this will need a significant rewrite anyway for the update.

Reply:
oh, then this is just oddly constructed.

Reply:
if you think its good, then go ahead and merge. Not terrible, just unusual structure from upstream

Reply:
I intentionally use make here, as it takes care of more things as `buildGoPackage` and I wanted to use their way of building and installing. While I could use `buildGoPackage` for some packages, it only complicates and brings in unnecessary complexity.